modules/light-clients/07-tendermint/types/update.go

```diff
@@ -180,7 +214,9 @@ func update(ctx sdk.Context, clientStore sdk.KVStore, clientState *ClientState,
 
 	// set context time as processed time as this is state internal to tendermint client logic.
 	// client state and consensus state will be set by client keeper
+	// set iteration key to provide ability for efficient ordered iteration of consensus states.
 	SetProcessedTime(clientStore, header.GetHeight(), uint64(ctx.BlockTime().UnixNano()))
+	SetIterationKey(clientStore, header.GetHeight())
```

Comment:
I think we need this to be set during the upgrade function right? We should add a test as well, but it seems to me and upgrade would cause pruning to break as the code currently stands